central/image/datastore/datastore_impl.go

```diff
@@ -72,35 +74,22 @@ func (ds *datastoreImpl) Count(ctx context.Context, q *v1.Query) (int, error) {
 func (ds *datastoreImpl) SearchImages(ctx context.Context, q *v1.Query) ([]*v1.SearchResult, error) {
 	defer metrics.SetDatastoreFunctionDuration(time.Now(), "Image", "SearchImages")
 
-	// TODO(ROX-29943): remove unnecessary calls to database
+	q = proto.Clone(q).(*v1.Query)
```

Comment:
why this instead of 
```
	if q == nil {
		q = pkgSearch.EmptyQuery()
	}

	// Clone the query and add select fields for SearchResult construction
	clonedQuery := q.CloneVT()
```

Reply:
done